### PR TITLE
feat(elb): update elb listener resource for new param

### PR DESCRIPTION
**Pull request by SabreKeyZ**
**What this PR does / why we need it**:
update elb listener resource for new param
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/elb/' TESTARGS='-ru
n TestAccElbV3Listener_with_port_ranges'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb/ -v -run TestAccElbV3Listener_basic -timeout 360m -parallel 4
=== RUN   TestAccElbV3Listener_basic
=== PAUSE TestAccElbV3Listener_basic
=== CONT  TestAccElbV3Listener_basic
--- PASS: TestAccElbV3Listener_basic (91.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       91.324s

make testacc TEST='./huaweicloud/services/acceptance/elb/' TESTARGS='-ru
n TestAccElbV3Listener_with_port_ranges'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb/ -v -run TestAccElbV3Listener_with_port_ranges -timeout 360m -parallel 4 
=== RUN   TestAccElbV3Listener_with_port_ranges 
=== PAUSE TestAccElbV3Listener_with_port_ranges
=== CONT  TestAccElbV3Listener_with_port_ranges
--- PASS: TestAccElbV3Listener_with_port_ranges (58.63s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       58.671s


```

### CODE REVIEW

File: docs/resources/elb_listener.md [hunk 1]
```diff
@@ -114,6 +114,21 @@ The following arguments are supported:
 
 * `tags` - (Optional, Map) The key/value pairs to associate with the listener.
 
+* `gzip_enable` - (Optional, Bool) Whether to enable gzip compression for a load balancer. The default value is **false**.
+  This parameter can be configured only for **HTTP**, **HTTPS**, and **QUIC** listeners.
+
+* `port_ranges` - (Optional, List, ForceNew) The port monitoring range (closed range), specify up to 10 port groups, each
+  group range must not overlap. This field can only be passed in when protocol_port is 0 or protocol_port is not passed
+  in. Only **TCP**, **UDP**, and **TCPSSL** listening support this field.
+  The [port_ranges](#ELB_port_ranges) structure is documented below.
```
Comment (houpeng80) [inline docs/resources/elb_listener.md]:
```suggestion
* `port_ranges` - (Optional, List, ForceNew) The port monitoring range (closed range), specify up to 10 port groups, each
  group range must not overlap. This field can only be passed in when `protocol_port` is **0** or empty. Only **TCP**, **UDP**, and **TCPSSL** listening support this field.
  The [port_ranges](#ELB_port_ranges) structure is documented below.
  Changing this creates a new listener.
```
put this param together with protocol_port

File: docs/resources/elb_listener.md [hunk 2]
```diff
@@ -38,12 +38,12 @@ The following arguments are supported:
...
   new listener.
 
-* `protocol_port` - (Required, Int, ForceNew) The port on which to listen for client traffic. Changing this creates a
-  new listener.
-
 * `loadbalancer_id` - (Required, String, ForceNew) The load balancer on which to provision this listener. Changing this
   creates a new listener.
 
+* `protocol_port` - (Optional, Int, ForceNew) The port used by the listener. The **QUIC** listener port cannot be 4789
+  or the same as the **UDP** listener port. If this parameter is set to 0, `port_ranges` is required.
```
Comment (houpeng80) [inline docs/resources/elb_listener.md]:
```suggestion
* `protocol_port` - (Optional, Int, ForceNew) The port used by the listener. The **QUIC** listener port cannot be **4789**
  or the same as the **UDP** listener port. If this parameter is set to **0**, `port_ranges` is required. Changing this creates a new listener.
```

File: docs/resources/elb_listener.md [hunk 3]
```diff
@@ -114,6 +114,21 @@ The following arguments are supported:
...
+  group range must not overlap. This field can only be passed in when protocol_port is 0 or protocol_port is not passed
+  in. Only **TCP**, **UDP**, and **TCPSSL** listening support this field.
+  The [port_ranges](#ELB_port_ranges) structure is documented below.
+
+<a name="ELB_port_ranges"></a>
+The `port_ranges` block supports:
+
+* `start_port` - (Optional, Int, ForceNew) Specifies the start port.
+
+* `end_port` - (Optional, Int, ForceNew) Specifies the end port.
```
Comment (houpeng80) [inline docs/resources/elb_listener.md]:
```suggestion
* `start_port` - (Required, Int, ForceNew) Specifies the start port. Changing this creates a new listener.

* `end_port` - (Required, Int, ForceNew) Specifies the end port. Changing this creates a new listener.
```

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go [hunk 4]
```diff
@@ -190,6 +191,34 @@ func ResourceListenerV3() *schema.Resource {
...
+						},
+						"end_port": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: true,
+							Computed: true,
+						},
+					},
+				},
+			},
```
Comment (houpeng80) [inline huaweicloud/services/elb/resource_huaweicloud_elb_listener.go]: put it together with protocol_port

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go [hunk 5]
```diff
@@ -56,8 +56,9 @@ func ResourceListenerV3() *schema.Resource {
 
 			"protocol_port": {
 				Type:     schema.TypeInt,
-				Required: true,
+				Optional: true,
 				ForceNew: true,
+				Computed: true,
```
Comment (houpeng80) [inline huaweicloud/services/elb/resource_huaweicloud_elb_listener.go]: AtLeastOneOf: []string{"protocol_port", "port_ranges"}

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go [hunk 6]
```diff
@@ -190,6 +191,34 @@ func ResourceListenerV3() *schema.Resource {
...
+							Optional: true,
+							ForceNew: true,
+							Computed: true,
+						},
+						"end_port": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: true,
+							Computed: true,
+						},
```
Comment (houpeng80) [inline huaweicloud/services/elb/resource_huaweicloud_elb_listener.go]:
```suggestion
						"start_port": {
							Type:     schema.TypeInt,
							Required: true,
							ForceNew: true,
						},
						"end_port": {
							Type:     schema.TypeInt,
							Required: true,
							ForceNew: true,
						},
```

File: docs/resources/elb_listener.md (hunk 1 above)
Comment (SabreKeyZ) [reply docs/resources/elb_listener.md]: done

File: docs/resources/elb_listener.md (hunk 2 above)
Comment (SabreKeyZ) [reply docs/resources/elb_listener.md]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go (hunk 5 above)
Comment (SabreKeyZ) [reply huaweicloud/services/elb/resource_huaweicloud_elb_listener.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go (hunk 4 above)
Comment (SabreKeyZ) [reply huaweicloud/services/elb/resource_huaweicloud_elb_listener.go]: done

File: docs/resources/elb_listener.md (hunk 3 above)
Comment (SabreKeyZ) [reply docs/resources/elb_listener.md]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go (hunk 6 above)
Comment (SabreKeyZ) [reply huaweicloud/services/elb/resource_huaweicloud_elb_listener.go]: done

**Pull request merged**